features/lorawan/lorastack/phy/loraphy_target.h

```diff
@@ -18,37 +18,50 @@
 #ifndef LORAPHY_TARGET
 #define LORAPHY_TARGET
 
+#define EU868 0
+#define AS923 1
+#define AU915 2
+#define CN470 3
+#define CN779 4
+#define EU433 5
+#define IN865 6
+#define KR920 7
+#define US915 8
+#define US915_HYBRID 9
+
 #ifdef MBED_CONF_LORA_PHY
- #if MBED_CONF_LORA_PHY      == 0
+ #if MBED_CONF_LORA_PHY      == EU868
   #include "lorawan/lorastack/phy/LoRaPHYEU868.h"
   #define LoRaPHY_region LoRaPHYEU868
- #elif MBED_CONF_LORA_PHY    == 1
+ #elif MBED_CONF_LORA_PHY    == AS923
   #include "lorawan/lorastack/phy/LoRaPHYAS923.h"
   #define LoRaPHY_region LoRaPHYAS923
- #elif MBED_CONF_LORA_PHY    == 2
+ #elif MBED_CONF_LORA_PHY    == AU915
   #include "lorawan/lorastack/phy/LoRaPHYAU915.h"
  #define LoRaPHY_region LoRaPHYAU915;
- #elif MBED_CONF_LORA_PHY    == 3
+ #elif MBED_CONF_LORA_PHY    == CN470
   #include "lorawan/lorastack/phy/LoRaPHYCN470.h"
   #define LoRaPHY_region LoRaPHYCN470
- #elif MBED_CONF_LORA_PHY    == 4
+ #elif MBED_CONF_LORA_PHY    == CN779
   #include "lorawan/lorastack/phy/LoRaPHYCN779.h"
   #define LoRaPHY_region LoRaPHYCN779
- #elif MBED_CONF_LORA_PHY    == 5
+ #elif MBED_CONF_LORA_PHY    == EU433
   #include "lorawan/lorastack/phy/LoRaPHYEU433.h"
   #define LoRaPHY_region LoRaPHYEU433
- #elif MBED_CONF_LORA_PHY    == 6
+ #elif MBED_CONF_LORA_PHY    == IN865
   #include "lorawan/lorastack/phy/LoRaPHYIN865.h"
   #define LoRaPHY_region LoRaPHYIN865
- #elif MBED_CONF_LORA_PHY    == 7
+ #elif MBED_CONF_LORA_PHY    == KR920
   #include "lorawan/lorastack/phy/LoRaPHYKR920.h"
   #define LoRaPHY_region LoRaPHYKR920
- #elif MBED_CONF_LORA_PHY    == 8
+ #elif MBED_CONF_LORA_PHY    == US915
   #include "lorawan/lorastack/phy/LoRaPHYUS915.h"
   #define LoRaPHY_region LoRaPHYUS915
- #elif MBED_CONF_LORA_PHY    == 9
+ #elif MBED_CONF_LORA_PHY    == US915_HYBRID
   #include "lorawan/lorastack/phy/LoRaPHYUS915Hybrid.h"
   #define LoRaPHY_region LoRaPHYUS915Hybrid
+ #else
+  #error "Unsupported region, check your configuration."
```

Comment:
Unfortunately this won't work, because any unknown identifier will be interpreted as 0, so it will pass the `#if MBED CONF_LORA_PHY == EU868` test.

You need the identifiers you recognise to be non-zero. Which is a bit unfortunate if you're trying to be backwards-compatible with having a meaning for 0.